Branch-SDK/Branch-SDK/BNCPreferenceHelper.m

```diff
@@ -542,6 +540,57 @@ - (NSInteger)getBranchViewCount:(NSString *)branchViewID {
     return count;
 }
 
+- (void)saveBranchAnalyticsData:(NSDictionary *)analyticsData {
+    if (_sessionID != nil) {
+        if (_savedAnalyticsData == nil) {
+            _savedAnalyticsData = [self getBranchAnalyticsData];
+        }
+        NSMutableArray *viewDataArray;
+        if ([_savedAnalyticsData objectForKey:_sessionID] != nil) {
+            viewDataArray = [_savedAnalyticsData objectForKey:_sessionID];
+        }
+        else {
+            viewDataArray = [[NSMutableArray alloc]init];
+            [_savedAnalyticsData setObject:viewDataArray forKey:_sessionID];
+        }
+        [viewDataArray addObject:analyticsData];
+        [self writeObjectToDefaults:KEY_BRANCH_ANALYTICAL_DATA value:_savedAnalyticsData];
+    }
+}
+
+- (void)clearBranchAnalyticsData {
+    [self writeObjectToDefaults:KEY_BRANCH_ANALYTICAL_DATA value:nil];
+    _savedAnalyticsData = nil;
+}
+
+- (NSMutableDictionary *)getBranchAnalyticsData {
+    NSMutableDictionary *analyticsDataObj;
+    if (_savedAnalyticsData != nil) {
+        analyticsDataObj  = _savedAnalyticsData;
+    } else {
```

Comment:
Noticed this pattern throughout if you wouldn't mind going and cleaning it up too! Thanks :)